napari/utils/transforms/_tests/test_transforms.py

```diff
@@ -240,3 +240,24 @@ def test_affine_matrix_inverse(dimensionality):
     np.testing.assert_almost_equal(
         transform.inverse.affine_matrix, np.linalg.inv(A)
     )
+
+
+def test_repeat_shear_setting():
+    """Test repeatedly setting shear with a lower triangular matrix."""
+    # Note this test is needed to check lower triangular
+    # decomposition of shear is working
+    mat = np.eye(3)
+    mat[2, 0] = 0.5
+    transform = Affine(shear=mat)
+    # Check shear decomposed into lower triangular
+    np.testing.assert_almost_equal(mat, transform.shear)
+
+    # Set shear to same value
+    transform.shear = mat
+    # Check shear still decomposed into lower triangular
+    np.testing.assert_almost_equal(mat, transform.shear)
+
+    # Set shear to same value
+    transform.shear = mat
+    # Check shear still decomposed into lower triangular
+    np.testing.assert_almost_equal(mat, transform.shear)
```

Comment:
Lovely. =) One thing I would say is that there is still a failure mode here, if somehow `Affine` overwrites `mat`. So I would do above: `transform = Affine(shear=mat.copy())`, and so on in the various segments. This is unlikely to be an error arising in the future but I'd still sleep more soundly with that code. I'll suggest it now.